Add user doc for SO tagging (#85557)
elastic/kibana: docs/management/managing-tags.asciidoc

[role="xpack"]
[[managing-tags]]
== Tags

The *tags* management section allows you to manage your tags, and assign them
to existing saved objects.

> Assigning tags to objects helps to categorize them, and allows to easily find

Reviewer: maybe `Assigning tags to objects helps to categorize, find, filter them...`?

the objects by filtering by tag(s) in the <<managing-saved-objects, Saved Objects management section>>
or from the other applications.

To access the tags management page, open the main menu, then click on *Stack Management > Tags*.

[role="screenshot"]
image::images/tags/tag-management-section.png[Tags management section]

[float]
=== Required permissions

> The `Tag Management` {kib} privilege is required to access the *Tags* management section.

Reviewer: It is a technical detail. Maybe we should provide it *after* the `Operations with tags` section to not overload a reader?

Author: You may be right. I followed the same order as the SO management documentation, but this specific doc/page seems outdated. @gchaps wdyt?

To add the privilege, open the menu, then click *Stack Management > Roles*.

Having the `read` privilege allows a user to:
- Access the `tags` management section
- Assign tags to saved objects he has `write` permission for

Having the `write` privilege allows a user to additionally:
- Create tags
- Delete tags
- Edit existing tags

> NOTE: having the `Tag Management` {kib} privilege is not required to
> see tags assigned on objects the user can see, or to filter objects by tags
> in {kib} applications or from the navigational search.

Reviewer 2: ```suggestion
NOTE: having the `Tag Management` {kib} privilege is not required to
view tags assigned on objects the user has `read` access to or to filter objects by tags
in {kib} applications or from the navigational search.
```

> NOTE: having `write` `Tag Management` privilege is not sufficient
> to assign tags to saved objects. The user also needs to have `write` permission
> on the objects he wants to assign tags to.

Reviewer 2: Let me know if this isn't accurate, but I think this is more specific to what we mean

```suggestion
NOTE: having `write` `Tag Management` privilege does not affect which saved objects
the user can assign tags to. The user needs to have `write` permission
on the objects they want to assign tags to.
```

[float]
[[settings-create-tag]]
=== Create a tag

> This action creates a new tag to be assigned to your saved objects.

Reviewer 2: ```suggestion
This action creates a new tag that can be assigned to saved objects.
```

To create a tag:

> . Click on the `create tag` button

Reviewer 2: ```suggestion
. Click on the `Create tag` button
```

. Choose the name and a color for the new tag

> . Click `save`

Reviewer 2: ```suggestion
. Click `Save`
```

[role="screenshot"]
image::images/tags/create-tag.png[Tag creation popin]

> NOTE: A tag's name is limited to 50 characters, and can only be composed of alphanumeric characters,
> `:`, `-`, `_` and whitespaces.

Reviewer 2: nit: maybe we should clarify what alphanumeric means for less technical users? Also do we not support other unicode characters for non-English users? (specifically, I'm thinking about the Chinese & Japanese users)

```suggestion
NOTE: A tag's name is limited to 50 characters, and can only be composed of alphanumeric characters (English letters and digits),
`:`, `-`, `_` and the space character.
```

[float]
[[settings-assign-tag]]
=== Manage a tag assignments to saved objects

> This action allows assigning or unassigning a tag to a list of saved object.

Reviewer 2: ```suggestion
This action allows assigning or unassigning a tag to a list of saved objects.
```

> This action allows assigning or unassigning a tag to a list of saved object.

Author: > You must have have `write` permission on the objects to which you assign the tag.

The results in the flyout are pre-filtered to only display objects that the user have `write` permission for. Should we still mention that?

> This action allows assigning or unassigning a tag to a list of saved object.

Reviewer 3: Yes

[role="screenshot"]
image::images/tags/manage-assignments-action.png[Assign action]

> To manage a tag's assignments:

Author: Moving the image between bullet points resets their number. When doing that, I got

```
1. Click the action (...) icon in the tag row, and then select the *Manage assignments* action.
<image>
1. Select the objects to which you want to assign or remove tags.
2. Click on *Save tag assignments*.
```

> To manage a tag's assignments:

Reviewer 3: If you put a "+" sign after the step, it should work. Like this:

Click the action (...) icon in the tag row, and then select the *Manage assignments* action.
+
[role="screenshot"]
image::images/tags/manage-assignments-action.png[Assign action]

> . Click on the `manage assignments` action in the tag's row

Reviewer 2: ```suggestion
. Click on the `Manage assignments` action in the tag's row
```

. Search for the saved objects you want to assign or unassign to the tag
. Add or remove the objects from the assignments
. Click on `Save tag assignments`

[role="screenshot"]
image::images/tags/manage-assignments-flyout.png[Assign flyout]

NOTE: it is also possible to assign multiple tags to objects at once, by selecting them in the listing

> and then using the `Manage tag assignments` action from the action menu.

Reviewer 3: ```suggestion
and then select *Manage tag assignments* action from the *selected tag* menu.
```

> NOTE: it is also possible to assign multiple tags to objects at once, by selecting them in the listing
> and then using the `Manage tag assignments` action from the action menu.

Reviewer 2: ```suggestion
NOTE: it is also possible to assign multiple tags to objects at once, by selecting them in the list
and then using the `Manage tag assignments` action from the action menu above the table.
```

[float]
[[settings-delete-tag]]
=== Delete a tag

> This action deletes a tag, and unassign it from any saved objects it was currently
> assigned to.

Reviewer 2: ```suggestion
This action deletes a tag and unassigns it from any saved objects it is currently
assigned to.
```

[role="screenshot"]
image::images/tags/delete-tag-action.png[Delete action]

To delete a tag:

> . Click on the `delete` action in the tag's row

Reviewer 2: ```suggestion
. Click on the `Delete` action in the tag's row
```

> . Click `delete` in the confirmation modal

Reviewer 2: ```suggestion
. Click `Delete` in the confirmation modal
```

[role="screenshot"]
image::images/tags/delete-tag-modal.png[Delete modal]

> NOTE: it is also possible to delete multiple tags at once, by selecting them in the listing

Reviewer 3: ```suggestion
TIP: To delete multiple tags at once, select their checkboxes in the list view,
```

> and then using the `delete` action from the action menu.

Reviewer 3: ```suggestion
and then select *Delete* action from the *selected tags* menu.
```

> NOTE: it is also possible to delete multiple tags at once, by selecting them in the listing
> and then using the `delete` action from the action menu.

Reviewer 2: ```suggestion
NOTE: it is also possible to delete multiple tags at once by selecting them in the listing
and clicking the `Delete` action from the action menu.
```

> and then using the `delete` action from the action menu.

Author: As this specific page is about the tags management section, I would expect documentation on usages in navSearch / other app to be in their respective documentation pages. E.g the 'navigational search' doc should describe how to filter by tag when using it. wdyt?

> and then using the `delete` action from the action menu.

Reviewer 3: @pgayvallet sounds good.  I'll tag you on a gdoc I'm writing to and you can add the content there.

[role="xpack"]
[[managing-tags]]
== Tags

Tags enable you to categorize your saved objects. You can then easily filter for related objects based on shared tags.

To begin, open the main menu, click *Stack Management*, then click *Tags*.

[role="screenshot"]
image::images/tags/tag-management-section.png[Tags management section]

[float]
=== Required permissions

Access to *Tags* requires the `Tag Management` {kib} privilege. To add the privilege, open the menu, 
click *Stack Management*, then click *Roles*.

In addition:

* The `read` privilege allows you to assign tags to the saved objects for which you have write permission.
* The `write` privilege enables you to create, edit, and delete tags.

> NOTE: having the `Tag Management` {kib} privilege is not required to

Reviewer 3: ```suggestion
NOTE: Having the `Tag Management` {kib} privilege is not required to
```

view tags assigned on objects the user has `read` access to, or to filter objects by tags
in {kib} applications or from the navigational search.

[float]
[[settings-create-tag]]
=== Create a tag

Create a tag to assign to your saved objects.

. Click *Create tag*.

> . Enter a name and select a color for the new tag.

Reviewer 3: We lost the thought about limitations on names.

. Enter a name and select a color for the new tag.
+ 
The name can include the characters ..... and cannot be longer than 50 characters.

If you use a "+" after the step, the numbering should be ok.

. Click *Create tag*.

[role="screenshot"]
image::images/tags/create-tag.png[Tag creation popin]

[float]
[[settings-assign-tag]]
=== Assign a tag to saved objects

Assign or remove tags to one or more saved objects.

. Click the action (...) icon in the tag row, and then select the *Manage assignments* action.
. Select the objects to which you want to assign or remove tags.
. Click on *Save tag assignments*.

[role="screenshot"]
image::images/tags/manage-assignments-flyout.png[Assign flyout]

> TIP: To assign multiple tags to objects at once, select their checkboxes

Reviewer 3: although I had to add another image, what do you think about showing a screenshot with the checkboxes checked?
